### PR TITLE
Code for get/api/articles

### PR DESCRIPTION
**Pull request by mejaz4**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ afterAll(() => {
 
 beforeEach(() => seed(testData))
 
-describe('1: GET api/treasures', () => {
+describe('1: GET api/topics', () => {
```
Comment (dav3rid) [inline __tests__/app.test.js:13]: Take care when copying code from elsewhere 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -37,3 +37,27 @@ describe('1: GET api/treasures', () => {
...
       });
 })
+
+describe('2. GET /api/articles', () => {
+    test('status:200, responds with an array of articles objects', () => {
+      return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then(({ body: { articles } }) => {
+            console.log(articles, '<<treasure')
```
Comment (dav3rid) [inline __tests__/app.test.js]: Can remove this console log

File: __tests__/app.test.js [hunk 3]
```diff
@@ -37,3 +37,27 @@ describe('1: GET api/treasures', () => {
...
+                article_id: expect.any(Number),
+                topic: expect.any(String),
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                comment_count: expect.any(String),
+              })
+          });
+        });
+    })
+})
```
Comment (dav3rid) [inline __tests__/app.test.js:62]: Solid test!
No explicit need to check if articles is an instance of an array - you do this implicitly when invoking a `forEach`
It's not hurting anyone, but can be removed

File: controllers/controller.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+})
+}
+
+const getArticles = (req, res, next) => {
+    // console.log("here")
+    selectArticles().then((articles) => {
+        // console.log("here??")
+        res.status(200).send({articles})
+    })
+}
```
Comment (dav3rid) [inline controllers/controller.js:15]: Controller looks good - however, it doesn't really belong in a file called `controllers.topics.js`
consider extracting out and removing the console logs

File: models/model.topics.js [hunk 5]
```diff
@@ -7,4 +7,18 @@ const selectTopics = () => {
...
\ No newline at end of file
+
+const selectArticles = () => {
+return db.query(`SELECT articles.*, COUNT(comments.comment_id) AS comment_count FROM articles
+LEFT JOIN comments 
+ON articles.article_id = comments.article_id
+GROUP BY articles.article_id
+ORDER BY created_at DESC;`)
+.then((result) => {
+    console.log(result.rows, "<< here are articles")
```
Comment (dav3rid) [inline models/model.topics.js]: another console log to strip out

File: models/model.topics.js [hunk 6]
```diff
@@ -7,4 +7,18 @@ const selectTopics = () => {
     })
 }
 
-module.exports = {selectTopics}
\ No newline at end of file
+
+const selectArticles = () => {
+return db.query(`SELECT articles.*, COUNT(comments.comment_id) AS comment_count FROM articles
```
Comment (dav3rid) [inline models/model.topics.js]: Don't want to select `articles.*` here as we don't want the body. Your test above for this looks sound but you are still sending it anyway. Select instead the specific columns you do want.
Similar feedback regarding file location too - should have an articles model file as well 👍

**Review by dav3rid — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js [hunk 7]
```diff
@@ -37,3 +37,26 @@ describe('1: GET api/treasures', () => {
...
       });
 })
+
+describe('2. GET /api/articles', () => {
+    test('status:200, responds with an array of articles objects', () => {
+      return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeInstanceOf(Array);
```
Comment (RichardNeat) [inline __tests__/app.test.js:47]: This is the line Dave was talking about, we don't need to explicitly check for an array here as we are later using a forEach and checking the length so it must be an array right?

**Review by RichardNeat — COMMENTED** [review]
Hi Musa - as Dave mentioned please do have a look at your file structure. Consider file names such as controller.articles.js which is logical as these are your article controller functions. Same for models. 'Topics' isn't relevant to your code so shouldn't be in the filename. Look at your test for an array as mentioned too.
Please also make sure you are only merging your branches into main when they are approved.
Once you have made the above changes (priority is file structure) can you drop me a message to confirm please?

File: __tests__/app.test.js (hunk 7 above)
Comment (mejaz4) [reply __tests__/app.test.js:47]: Aren't you checking if it is an array before we used forEach? forEach could also be used for an object no?

Comment (RichardNeat) [reply __tests__/app.test.js:47]: forEach is an array method

**Pull request merged**